build.gradle.kts

```diff
@@ -112,28 +134,25 @@ pluginBundle {
 }
 
 tasks {
-    val sampleProjects by creating(GradleBuild::class) {
+    val integrationTests by creating(GradleBuild::class) {
         dir = File("${project.rootDir}/tests")
         tasks = listOf("clean", "build")
     }
-    sampleProjects.dependsOn("publishToMavenLocal")
-    "test" { finalizedBy(sampleProjects) }
+    integrationTests.dependsOn("publishToMavenLocal")
+    "test" { finalizedBy(integrationTests) }
     named<Test>("test") {
         testLogging.exceptionFormat = TestExceptionFormat.FULL
     }
```

Comment:
This part looks a bit odd. It's not clear why can't we configure the task inside one closure. Also, I do understand this part is heavily inspired by a third-party plugin.